webapp/src/components/shareBoard/shareBoardLoginButton.tsx

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useCallback} from 'react'
+import {FormattedMessage} from 'react-intl'
+import {generatePath, useRouteMatch, useHistory} from 'react-router-dom'
+
+import Button from '../../widgets/buttons/button'
+import TelemetryClient, {TelemetryActions, TelemetryCategory} from '../../telemetry/telemetryClient'
+import {Utils} from '../../utils'
+
+import './shareBoardLoginButton.scss'
+
+const ShareBoardLoginButton = () => {
+    const match = useRouteMatch<{teamId: string, boardId: string, viewId?: string, cardId?: string}>()
+    const history = useHistory()
+
+    let redirectQueryParam = 'r=' + encodeURIComponent(generatePath('/:boardId?/:viewId?/:cardId?', match.params))
+    if (Utils.isFocalboardLegacy()) {
+        redirectQueryParam = 'redirect_to=' + encodeURIComponent(generatePath('/boards/team/:teamId/:boardId?/:viewId?/:cardId?', match.params))
+    }
+    const loginPath = '/login?' + redirectQueryParam
+
+    const onLoginClick = useCallback(() => {
+        TelemetryClient.trackEvent(TelemetryCategory, TelemetryActions.ShareBoardLogin)
+        if (Utils.isFocalboardLegacy()) {
+            location.assign(loginPath)
```

Comment:
Need to use `location.assign` because it needs to replace the entire path. A leading slash on the parameter causes that (so don't need the full URL). Without the leading slash, the path is appended to the base path. (like `history.push`)